doc/man3/BIO_sendmmsg.pod

```diff
@@ -0,0 +1,202 @@
+whether an operation should be retried; see below.
+
+The support for concurrent use extends to BIO_sendmmsg() and BIO_recvmmsg()
+only, and no other function may be called on a given BIO while any call to
+BIO_sendmmsg() or BIO_recvmmsg() is in progress, or vice versa.
+
+BIO_dgram_set_local_addr_enable() and BIO_dgram_get_local_addr_enable() control
+whether local address support is enabled. To enable local address support, call
+BIO_dgram_set_local_addr_enable() with an argument of 1. The call will fail if
+local address support is not available for the platform.
+BIO_dgram_get_local_addr_enable() retrieves the value set by
+BIO_dgram_set_local_addr_enable().
+
+BIO_dgram_get_local_addr_cap() determines if the B<BIO> is capable of supporting
+local addresses.
+
+BIO_err_is_non_fatal() determines if a packed error code represents an error
+which is transient in nature.
+
+=head1 NOTES
+
+Some implementations of the BIO_sendmmsg() and BIO_recvmmsg() BIO methods might
+always process at most one message at a time, for example when OS-level
+functionality to transmit or receive multiple messages at a time is not
+available.
+
+=head1 RETURN VALUES
+
+On success, the functions BIO_sendmmsg() and BIO_recvmmsg() return 1 and write
+the nummber of messages successfully processed (which need not be nonzero) to
+I<msgs_processed>. Where a positive value n is written to I<msgs_processed>, all
+entries in the B<BIO_MSG> array from 0 through n-1 inclusive have their
+I<data_len> and I<flags> fields updated with the results of the operation on
+that message. If the call was to BIO_recvmmsg() and the I<peer> or I<local>
+fields of that message are non-NULL, the B<BIO_ADDR> structures they point to
+are written with the relevant address.
+
+On failure, the functions BIO_sendmmsg() and BIO_recvmmsg() return 0 and write
+zero to I<msgs_processed>. Thus I<msgs_processed> is always written regardless
+of the outcome of the function call.
+
+If BIO_sendmmsg() and BIO_recvmmsg() fail, they always raise an error using
+L<ERR_raise(3)>. Any error may be raised, but the following in particular may be
+noted:
+
+=over 2
+
+=item B<BIO_R_LOCAL_ADDR_NOT_AVAILABLE>
+
+The I<local> field was set to a non-NULL value, but local address support is not
+available or not enabled on the BIO.
+
+=item B<BIO_R_UNSUPPORTED_METHOD>
+
+The BIO_sendmmsg() or BIO_recvmmsg() method is not supported on the BIO.
+
+=item Socket errors
+
+OS-level socket errors are reported using an error with library code
+B<ERR_LIB_SYS>; for a packed error code B<errcode> where
+C<ERR_SYSTEM_ERROR(errcode) == 1>, the OS-level socket error code can be
+retrieved using C<ERR_GET_REASON(errcode)>. The packed error code can be
+retrieved by calling L<ERR_peek_last_error(3)> after the call to BIO_sendmmsg()
+or BIO_recvmmsg() returns 0. Whether the error is transient can be determined by
+passing the packed error code to BIO_err_is_non_fatal().
```

Comment:
How would a non-socket based custom BIO implementation signal a retry (e.g. a mem BIO)?

Reply:
Via the same means? This doesn't rely on OS functionality so there's no reason alternative BIO implementations can't signal the same way.

Reply:
`BIO_err_is_non_fatal()` checks whether the error is `ERR_LIB_SYS` and if so what sys error code it is. It seems a bit weird for a mem bio to be indicating a syscall error when it hasn't made any syscalls?